apis/core/lib/domain/csv-source/upload.ts

```diff
@@ -37,6 +40,22 @@ export async function uploadFile({
         .addOut(schema.contentUrl, $rdf.namedNode(upload.Location))
     })
 
+  try {
+    const fileStream = await loadFile(key) as Readable
+    const head = await loadFileHeadString(fileStream)
+    const { dialect, header } = await sniffParse(head)
+
+    csvSource
+      .addOut(csvw.dialect, csvDialect => {
+        csvDialect.addOut(csvw.quoteChar, dialect.quote)
```

Comment:
One last thing: please make the dialect a named node:

```suggestion
      .addOut(csvw.dialect, id.dialect(csvSource), csvDialect => {
        csvDialect.addOut(csvw.quoteChar, dialect.quote)
```

by appending /dialect to source's URI in id.dialect